build/pipelines/templates/pgo-build-and-publish-nuget-job.yml

```diff
@@ -0,0 +1,55 @@
+# From our friends at MUX: https://github.com/microsoft/microsoft-ui-xaml/blob/main/build/AzurePipelinesTemplates/MUX-BuildAndPublishPGONuGet-Job.yml
+
+parameters:
+  dependsOn: ''
+  pgoArtifact: PGO
+
+jobs:
+- job: BuildAndPublishPGONuGet
+  dependsOn: ${{ parameters.dependsOn }}
+  pool:
+    vmImage: 'windows-2019'
+  variables:
+    artifactsPath: $(Build.SourcesDirectory)\Artifacts
+    pgoToolsPath: $(Build.SourcesDirectory)\tools\PGODatabase
+    nuspecPath: $(pgoToolsPath)\NuSpecs
+    nuspecFilename: PGO.nuspec
+
+  steps:
+  - task: DownloadBuildArtifacts@0
+    inputs:
+      artifactName: ${{ parameters.pgoArtifact }}
+      downloadPath: $(artifactsPath)
+
+  - task: CopyFiles@2
+    displayName: 'Copy pgd files to NuGet build directory'
+    inputs:
+      sourceFolder: $(artifactsPath)\${{ parameters.pgoArtifact }}
+      contents: '**\*.pgd'
+      targetFolder: $(nuspecPath)\tools
+
+  - task: powershell@2
+    displayName: 'Generate NuSpec file'
+    inputs:
+      targetType: filePath
+      filePath: $(pgoToolsPath)\generate-nuspec.ps1
+      workingDirectory: $(pgoToolsPath)
+      arguments: $(nuspecPath)\$(nuspecFilename).template $(nuspecPath)\$(nuspecFilename)
+
+  - task: CmdLine@1
+    displayName: 'Build NuGet package'
+    inputs:
+      filename: '$(Build.SourcesDirectory)\tools\NugetWrapper.cmd'
```

Comment:
we can probably just use a nuget pack step unless we really need to run through a wrapper for some reason..

Reply:
@bartekk8, do you know why this went this way instead of using the step?

Reply:
Moved to step...